src/resources/affinity.js

```diff
@@ -48,15 +48,23 @@ const parsePreferredAffinity = R.map(x => {
 })
 const parseRequiredAffinity = R.compose(removeWeight, parsePreferredAffinity)
 
+const createSelfResolver = (name) => R.lift(R.when(
+  R.equals('self'), R.always({ name })
```

Comment:
I would do this with just standard JS array methods, or comment the heck of this.  if you don't know ramda this is an entirely alien construct.

Reply:
:alien: to me as well

Reply:
Commented. Does that clear it up at all?

Reply:
Better!